actix-web/tests/compression.rs

```diff
@@ -96,7 +96,7 @@ async fn negotiate_encoding_gzip() {
 
     let req = srv
         .post("/static")
-        .insert_header((header::ACCEPT_ENCODING, "gzip,br,zstd"))
+        .insert_header((header::ACCEPT_ENCODING, "gzip"))
```

Comment:
Not too happy with this. Should I add quality specifiers with lesser quality so gzip does get selected despite others specified?

Reply:
Doing so would make these tests non-representative of real browsers requests.

I'll have a mess about with this PR and update soon.

Reply:
Okay, I've tweaked the implementation and tests. LMK what you think!

Reply:
Looks good to me.